flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/ServerIdRange.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.mysql.source.config;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * This class defines a range of server id. The boundaries of the range are inclusive.
+ *
+ * @see MySqlSourceOptions#SERVER_ID
+ */
+public class ServerIdRange implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    /** Start of the range (inclusive). */
+    private final int startServerId;
+
+    /** End of the range (inclusive). */
+    private final int endServerId;
+
+    public ServerIdRange(int startServerId, int endServerId) {
+        this.startServerId = startServerId;
+        this.endServerId = endServerId;
+    }
+
+    public int getStartServerId() {
+        return startServerId;
+    }
+
+    public int getEndServerId() {
+        return endServerId;
+    }
+
+    public int getServerId(int subTaskId) {
+        checkArgument(subTaskId >= 0, "Subtask ID %s shouldn't be a negative number.", subTaskId);
+        if (subTaskId > getNumberOfServerIds()) {
+            throw new IllegalArgumentException(
+                    String.format(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: subTaskId (declared at line 54). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 58. Consider formatting this data with &#39;%d&#39; instead.